### PR TITLE
Changes for adding Datadog provider

### PR DESCRIPTION
**Pull request by lawwong1**
Replace signalfx by datadog as the metric provider.

Design doc: https://paypal.atlassian.net/wiki/spaces/GE/pages/972098897/Trimaran+Load-Watcher+to+support+Datadog+client

TEST DONE:

verify datadog metrics pulled to load-watcher in dev52 and dcg01 cluster with json payload to datadog api
unit tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-COMMONENVVAR=GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(subst x86_64,amd64,$(patsubst i%86,386,$(shell uname -m)))
+COMMONENVVAR=GOOS=$(uname -s | tr A-Z a-z)
+GOARCHVAR=GOARCH=$(uname -m | tr A-Z a-z)
```
[the remark this answers is not in the document]
Comment (lawwong1) [reply Makefile:16]: Correct, got below error:
=> ERROR [stage-0 4/4] RUN make build                                                                                     0.2s
------
 > [stage-0 4/4] RUN make build:
0.214 GOOS=linux GOARCH=aarch64 CGO_ENABLED=0 go build -o bin/load-watcher main.go
0.216 go: unsupported GOOS/GOARCH pair linux/aarch64
0.217 make: *** [Makefile:24: build] Error 2

[the remark this answers is not in the document]
Comment (lenhattan86) [reply Makefile:16]: ok.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,6 @@
-FROM golang:1.16
+FROM dockerhub.paypalcorp.com/golang:1.22-bullseye AS builder
+
+ENV GOPROXY=https://artifactory.g.devqa.gcp.dev.paypalinc.com/artifactory/api/go/go
```
[the remark this answers is not in the document]
Comment (lawwong1) [reply Dockerfile]: Need ENV GOPROXY=https://artifactory.g.devqa.gcp.dev.paypalinc.com/artifactory/api/go/go, else would got below error in docker build --build-arg BUILD_SPEC=arm64 -t load-watcher:v0.3.0 . 

68.10 pkg/watcher/watcher.go:35:2: github.com/francoispqt/gojay@v1.2.13: Get "https://proxy.golang.org/github.com/francoispqt/gojay/@v/v1.2.13.zip": tls: failed to verify certificate: x509: certificate signed by unknown authority
68.10 pkg/watcher/watcher.go:36:2: github.com/sirupsen/logrus@v1.6.0: Get "https://proxy.golang.org/github.com/sirupsen/logrus/@v/v1.6.0.zip": tls: failed to verify certificate: x509: certificate signed by unknown authority

File: pkg/watcher/internal/metricsprovider/datadog.go [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+	clusterName    string
+}
+
+// This method creates a new datadog client based on the environment variables
+func NewDatadogClient(opts watcher.MetricsProviderOpts) (watcher.MetricsProviderClient, error) {
+	if opts.Name != watcher.DatadogClientName {
+		return nil, fmt.Errorf("metric provider name should be %v, found %v", watcher.DatadogClientName, opts.Name)
+	}
+	tlsConfig := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: opts.InsecureSkipVerify}, // TODO(aqadeer): Figure out a secure way to let users add SSL certs
```
[the remark this answers is not in the document]
Comment (lawwong1) [reply pkg/watcher/internal/metricsprovider/datadog.go]: Done and checked in.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (lawwong1) [reply Makefile]: I checked in new changes in Makefile and Dockerfile to properly parameterized build for different architecture.

**Review by lenhattan86 — APPROVED** [review]

**Pull request merged**